igor-web/src/main/groovy/com/netflix/spinnaker/igor/scm/stash/client/StashMaster.groovy

```diff
@@ -44,9 +47,20 @@ class StashMaster extends AbstractScmMaster {
     }
   }
 
-  String getTextFileContents(String projectKey, String repositorySlug, String path, String at) {
+  String getTextFileContents(String projectKey, String repositorySlug, String path, String ref) {
     try {
-      return stashClient.getTextFileContents(projectKey, repositorySlug, path, at).toTextContents()
+      String contents = ""
+      boolean lastPage = false
+      int start = 0
+      while (!lastPage) {
+        log.debug("Retrieving text file contents from project: $projectKey, repo: $repositorySlug, path: $path, ref: $ref, start: $start")
+        TextLinesResponse response = stashClient.getTextFileContents(
+          projectKey, repositorySlug, path, ref, DEFAULT_PAGED_RESPONSE_LIMIT, start)
+        lastPage = response.isLastPage
+        start = response.start + response.size
+        contents += response.toTextContents() + "\n"
```

Comment:
It's a bit more efficient to use a [StringBuilder](https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html) instead of repeated string conatenation in a loop.